src/manifest.c

```diff
@@ -801,6 +803,17 @@ struct list *create_update_list(struct manifest *server)
 			}
 		}
 
+		/* Delta manifests don't include unchanged files in the FROM file list,
+		 * so files in the TO file list without a peer are not new when the file
+		 * exists on the system. */
+		filename = mk_full_filename(path_prefix, file->filename);
```

Comment:
it looks like this comment doesn't make sense anymore after Matthews comment

Reply:
My suggestion isn't going to work. If an old bundle is included in a new version of a new bundle, the version of the file will be older than current version and it will be installed anyway.

I was trying to understand the best approach to that and I wrote a patch, but I still have 2 tests failing. https://github.com/otaviobp/swupd-client/commit/fcbf00934f13fde580470cbaad7ce7e23a93c380

Reply:
I think we can solve this case by doing the following:

1. Add a variable to the file struct that is set to true when a file is included

2. Initialize variable for loaded files to false

3. Mark files pulled in by included bundles true (in add_subscriptions).

4. Check for included files with an older version than the current.

Reply:
Since we can't use delta manifests to decide information about file updates, I no longer see a need for the link_manifests function. I think we could consolidate the file statistic counting to create_update_list and remove the link_manifests function.